### PR TITLE
Sequelize v5-Beta+ Compatibility fixes

### PR DESCRIPTION
**Pull request by Bostrong**
Hi There,

I really needed a working version of this with Sequelize v5-beta as I was relying on another branch of Sequelize which was is in sync with v5-beta master.

For refs sake that branch of Sequelize is ..
https://github.com/indream/sequelize/tree/postgres-on-conflict 

I sincerely hope this branch which supports native PostGres upserts with no conflict is merged some day as it is quite a deal breaker.

In any case it appears to work for the most part ... however I am not sure if it is entirely logically correct (might have hacked at a few bits).

Feel free to merge or re-write.

Cheers.

### CODE REVIEW

File: lib/hooksUniversal.js [hunk 1]
```diff
@@ -45,8 +51,13 @@ module.exports = function(Sequelize) {
 				field.hierarchy = true;
 			});
 
-			// if hierarchy set, init hierarchy
-			if (hierarchy) model.isHierarchy(hierarchy);
+			try{
```
Comment (overlookmotel) [inline lib/hooksUniversal.js]: What is the point of this change since the error just gets thrown again anyway?

File: lib/patches.js [hunk 2]
```diff
@@ -21,33 +21,38 @@ module.exports = function(Sequelize) {
...
          */
         underscoredIf: {
-          '^2.0.0': Sequelize.Utils._.underscoredIf,
-          '>=3.0.0': Sequelize.Utils.underscoredIf
+          '^2.0.0': Sequelize.Utils._? Sequelize.Utils._.underscoredIf : null,
+          '>=3.0.0 || >=5.0.0-beta': Sequelize.Utils.underscoredIf
         },
+        // For some reason in v5 uppercaseFirst is missing ... assume pluralize will do the job instead
+        uppercaseFirst: {
+            '>=5.0.0-beta': Sequelize.Utils.pluralize,
```
Comment (overlookmotel) [inline lib/patches.js]: `.pluralize()` does something else. I think this needs an explicit function e.g.:

```js
function(str) {
  return str.slice(0, 1).toUpperCase() + str.slice(1);
}
```

**Comment by overlookmotel** [comment]
Hi @Bostrong.

I've had a quick look. I've made a couple of comments on your changes, but to make it easier to review can you please remove all the extra line breaks you've added to reduce the diff?

Also, the additional functions you've added in `lib/utils.js` should be in `lib/patches.js` as they relate to different behavior across difference Sequelize versions.

If you can make those changes, I'll have another look.

**Comment by overlookmotel** [comment]
Also, can you please rebase onto current master? I'd like to merge your PR when it's ready rather than re-write myself.

**Comment by jasonleunghk01** [comment]
any updates on this pr 🙏

**Comment by AmbroiseVuaridel** [comment]
Any update on this ?

**Comment by ccckblaze** [comment]
the v5 version is about to release, can you help us out, thx
@overlookmotel

File: lib/errors.js [hunk 3]
```diff
@@ -28,6 +28,9 @@ module.exports = function(Sequelize) {
 		},
 		'>=4.0.0': function() {
 			return require('./errorsEs6')(Sequelize);
+		},
```
Comment (rusekr) [inline lib/errors.js]: semver-select will select >=4.0.0 in case of sequelize5 because it selects first choise that matches according to documentation

**Comment by AmbroiseVuaridel** [comment]
is this branch working fine ? can someone merge it ?

**Comment by rusekr** [comment]
using this in production https://github.com/overlookmotel/sequelize-hierarchy/pull/161 :)

**Comment by AmbroiseVuaridel** [comment]
> using this in production #161 :)
@rusekr 
Thanks! I get "Utils.uppercase First is not a function" in my sequelize init file were i create all my models, rings a bell ?

**Comment by rusekr** [comment]
in pr161 should not be this error. Utils from  sequeilize replaced with local utils

**Comment by ccckblaze** [comment]
it seems just need one more shot

**Comment by overlookmotel** [comment]
v2.0.0 just released which includes Sequelize v5 support.

**Pull request closed without merging**